crates/base/src/deno_runtime.rs

```diff
@@ -389,12 +468,25 @@ impl DenoRuntime {
             );
         }
 
+        if is_user_worker {
+            js_runtime.v8_isolate().add_gc_prologue_callback(
```

Comment:
@nyannyacha This callback would run before garbage collector would start?

Reply:
@laktek 
This callback is called just before V8 attempts to garbage collect.[^1][^2]

[^1]:https://source.chromium.org/chromium/chromium/src/+/main:v8/src/heap/heap.cc;l=1774?q=heap.cc
[^2]:https://source.chromium.org/chromium/chromium/src/+/main:v8/src/heap/heap.cc;l=1833?q=heap.cc